sa/_db/migrations/20170423163751_CreateAuthorizationsTable.sql

```diff
@@ -0,0 +1,31 @@
+
+-- +goose Up
+-- SQL in section 'Up' is executed when this migration is applied
+CREATE TABLE `authorizations` (
+  `id` bigint(20) NOT NULL AUTO_INCREMENT,
+  `token` mediumblob,
+  `identifierValue` varchar(255) NOT NULL,
+  `identifierType` int NOT NULL,
+  `registrationID` bigint(20) NOT NULL,
+  `status` int NOT NULL,
+  `expires` datetime NOT NULL,
+  `allowedChallenges` tinyint,
+  `validated` datetime NOT NULL,
+  `validationRecord` mediumblob,
+  `error` mediumblob,
+  `thumbprint` mediumblob,
+
+  PRIMARY KEY (`id`),
+  UNIQUE KEY `identifier_key` (`identifierType`, `identifierValue`),
+
+  KEY `registrationID_status_expires_idx` (`registrationID`, `status`, `expires`),
+  CONSTRAINT `registrationID_status_expires_idx` FOREIGN KEY (`registrationID`) REFERENCES `registrations` (`id`) ON DELETE NO ACTION ON UPDATE NO ACTION,
```

Comment:
I think you may have either misunderstood my comment, or typoed, but your most recent revision removed the two `KEY` statements that we want. I'm looking for this:

```
  KEY `registrationID_status_expires_idx` (`registrationID`, `status`, `expires`),
  KEY `identifierType_identifierValue_status_registrationID_expires_idx` (`identifierType`, `identifierValue`, `status`, `registrationID`, `expires`),

  CONSTRAINT `regId_authorizations`  FOREIGN KEY (`registrationID`) REFERENCES `registrations` (`id`)
```

That is, the `CONSTRAINT` is a completely separate stanza from the `KEY`s. Also, we don't need the `KEY regId_authorizations_idx (registrationID)` you added, since that is covered by the other `KEY`s.